### PR TITLE
Add refresh ttls

### PR DESCRIPTION
**Pull request by tom-hadlaw-hs**
Background:
Currently dynamic credentials are always (attempted) to be renewed ~10 minutes before they are set to expire. For example, if a AWS STS token token is set to expire after 60 minutes,  the periodic check will notice that it is set to expire prior to the next heartbeat and will attempt to renew the credentials.

This creates a worst case scenario where if vault goes down, existing dynamic credentials may only be good for a few more minutes (<10). In this case it may be better to generate credentials that exceed their desired usage. 

Changes:
v1.2.0: 26-May-2021
 * Added --force-refresh-ttl which temporary credentials will optionally be renewed before their actual expiry.
 * Added --sts-ttl flag which lets you specify token ttl for aws tokens
 generated using sts.
 * Added buildVersion value when running version flag (`vault-ctrl-tool --version`)
 * Embedded version now uses semver that will correspond with tagged versions.

### CODE REVIEW

File: briefcase/aws_sts.go [hunk 1]
```diff
@@ -23,13 +22,43 @@ func (b *Briefcase) AWSCredentialExpiresBefore(awsConfig config.AWSType, expires
...
+		refreshExpiry = &exp
+		if refreshExpiry.After(expiry) {
+			b.log.Warn().Msgf("forceRefreshTTL is longer than the expiry of aws credentials")
+		}
+		b.log.Info().Time("expiry", expiry).Time("refreshTime", exp).Str("outputPath", awsConfig.OutputPath).
+			Msg("enrolling AWS credential")
+	} else {
+		b.log.Info().Time("expiry", expiry).Str("outputPath", awsConfig.OutputPath).
+			Msg("enrolling AWS credential")
+		refreshExpiry = nil
```
Comment (brandon-mcrae-hs) [inline briefcase/aws_sts.go:56]: _nit._ It's already `nil` in this code branch.

File: briefcase/briefcase.go [hunk 2]
```diff
@@ -4,10 +4,11 @@ import (
 	"context"
 	"encoding/json"
 	"errors"
-	"github.com/hootsuite/vault-ctrl-tool/v2/metrics"
 	"io/ioutil"
 	"time"
 
+	"github.com/hootsuite/vault-ctrl-tool/v2/metrics"
```
Comment (brandon-mcrae-hs) [inline briefcase/briefcase.go:10]: _nit._ Imports in this file should probably just be a group for std lib and a group for everything else.

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,10 @@
+v1.2.0: 26-May-2021
```
Comment (brandon-mcrae-hs) [inline CHANGELOG:1]: Should we also include some docs for how to use the flags and why you might want to in the docs section? https://github.com/hootsuite/vault-ctrl-tool/tree/master/docs

This is helpful but I like examples, I'm sure others would too.

Comment (tom-hadlaw-hs) [reply CHANGELOG:1]: 👍

File: briefcase/briefcase.go [hunk 4]
```diff
@@ -63,6 +65,7 @@ type LeasedAuthToken struct {
 type leasedAWSCredential struct {
 	AWSCredential config.AWSType `json:"role"`
 	Expiry        time.Time      `json:"expiry"`
+	RefreshExpiry *time.Time     `json:"refresh_expiry,omitempty"`
```
[the remark this answers is not in the document]
Comment (tom-hadlaw-hs) [reply briefcase/briefcase.go:68]: Yeah, I didn't want to write the refresh_expiry unless it was set.

File: CHANGELOG (hunk 3 above)
Comment (tom-hadlaw-hs) [reply CHANGELOG:1]: added some docs here docs/examples/force-refreshing-credentials.md

**Review by brandon-mcrae-hs — APPROVED** [review]

**Pull request merged**